src/main.cpp

```diff
@@ -692,7 +691,7 @@ unsigned int GetP2SHSigOpCount(const CTransaction& tx, const CCoinsViewCache& in
 
 
 
-bool CheckTransaction(const CTransaction& tx, CValidationState &state)
+bool CheckTransaction(const CTransaction& tx, CValidationState &state, uint64_t nMaxTxSize)
```

Comment:
Another possibility is to just separate MAX_TX_SIZE (as a constant) and use that without passing anything as parameter. In any case, why uint64_t and not size_t ?
I really don't understand your point about the check being ignored when you set that unsigned to -1...